src/log_capture.jl

```diff
@@ -86,18 +104,6 @@ function _redirect_logs(f, target::IO)
     redirect_stdio(f, stdout=colored_io, stderr=colored_io)
 end
 
-# A lock that helps to stagger prints to DEFAULT_STDOUT, e.g. when there are log messages
-# comming from distributed workers, reports for stalled tests and outputs of
-# `print_errors_and_captured_logs`.
-const LogCaptureLock = ReentrantLock()
-macro loglock(expr)
-    return :(@lock LogCaptureLock $(esc(expr)))
```

Comment:
Why don't we need this anymore? I thought we would, because on the coordinator we `@spawn start_and_manage_worker` for each worker, so even though we aren't running each testitem on their own process, the coordinator could still have multiple threads processing results / printing logs (but maybe i'm misunderstanding)

Reply:
The lock was only used it in three places: `log_running`, `log_finished` and `print_errors_and_captured_logs`. The first two happened sequentially on each worker and the last one sequentially on the coordinator. So the way this was setup was not really helping anything. We might want to use a mechanism like this for the spawned tasks reading messages from the workers, as you mentioned, but since we are currently reading the output from the worker process line by line a more sophisticated strategy would be needed (like reading all available buffered bytes from the process, not just a line, making sure we handle the case where the buffer is full).

I think let's revisit this if we notice problems in practice.

Reply:
Ugh, you are right... I forgot this changed with the worker PR. Let me try playing with it with :batched logs :+1:.

Reply:
I think it's on the safer side to keep it in... thanks for noticing!